pkg/prometheus/promcfg.go

```diff
@@ -1058,6 +1071,12 @@ func (cg *configGenerator) generateServiceMonitorConfig(
 		})
 	}
 
+	// Relabel job_name into a meta label
+	relabelings = append(relabelings, yaml.MapSlice{
+		{Key: "target_label", Value: "__meta_prometheus_job_name"},
```

Comment:
As per https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config, the `__meta` prefix is reserved for Prometheus service discovery. We need to use `__tmp` rather.

```suggestion
		{Key: "target_label", Value: "__tmp_prometheus_job_name"},
```